### PR TITLE
Adds DateTime Extension Support

### PR DESCRIPTION
**Pull request by muditchaudhary**
## Overview
Adds DateTime Extension Support ([RFC 80](https://github.com/strongdm/cedar-rfcs/blob/datetime-rfc/text/0080-datetime-extension.md))

## Changes
- Implements new `com.cedarpolicy.value.DateTime` class
  - Validates input dates against supported format specifications
  - Stores `DateTime` value in `String` representation

- Integrates `DateTime` support in serialization layer
  - Extends `ValueSerializer` to handle `DateTime` serialization
  - Adds `DateTime` deserialization support in `ValueDeserializer`

- Testing
  - Adds unit tests covering `DateTime` validation, serialization, and edge cases
  - Includes integration tests to verify end-to-end `DateTime` functionality

- Updates CHANGELOG

## Example Usage
```java
Set<Entity> entities = new HashSet<>();
String principalId = "alice";
Map<String, Value> principalAttributes = new HashMap<>();
principalAttributes.put("DOB", new DateTime("2000-01-01"));
Set<EntityUID> principalParents = new HashSet<>();
Entity principal = new Entity(new EntityUID(principalType, principalId), principalAttributes, principalParents);
entities.add(principal);

String actionId = "view";
Map<String, Value> actionAttributes = new HashMap<>();
Set<EntityUID> actionParents = new HashSet<>();
Entity action = new Entity(new EntityUID(actionType, actionId), actionAttributes, actionParents);
entities.add(action);

String resourceId = "photo.jpg";
Map<String, Value> resourceAttributes = new HashMap<>();
Set<EntityUID> resourceParents = new HashSet<>();
var resource = new Entity(new EntityUID(resourceType, resourceId), resourceAttributes, resourceParents);
entities.add(resource);

String p = "permit(\n"
        + "principal==" + principal.getEUID().toString() + ",\n"
        + "action==" + action.getEUID().toString() + ",\n"
        + "resource==" + resource.getEUID().toString() + "\n"
        + ") when {principal.DOB > datetime(\"1999-01-01\")};";
                
final String policyId = "ID0";
Policy policy = new Policy(p, policyId);
Set<Policy> policies = new HashSet<>();
policies.add(policy);
PolicySet policySet = new PolicySet(policies);
Map<String, Value> currentContext = new HashMap<>();
AuthorizationRequest request = new AuthorizationRequest( principal, action, resource, currentContext);
```

### CODE REVIEW

File: CedarJava/src/main/java/com/cedarpolicy/value/DateTime.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        DateTime dateTime1 = (DateTime) o;
+        return dateTime.equals(dateTime1.dateTime);
```
[the remark this answers is not in the document]
Comment (mark-creamer-amazon) [reply CedarJava/src/main/java/com/cedarpolicy/value/DateTime.java:111]: Hmm, I'd expect we'd want a semantic value match here, right? Since in Cedar policies, `datetime("2024-10-15") == datetime("2024-10-15T11:35:00Z")` would be true.

They can and should keep their particular string representations, but the dates should be equivalent?

[the remark this answers is not in the document]
Comment (muditchaudhary) [reply CedarJava/src/main/java/com/cedarpolicy/value/DateTime.java:111]: Sure. I can make that change. I wasn't sure if we want them to be semantically equal as well.

File: CedarJava/src/main/java/com/cedarpolicy/value/DateTime.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            throw new IllegalArgumentException(
+                    "Input string is not a valid DateTime format: " + dateTime);
+        }
+        this.dateTime = dateTime;
+    }
+
+    /** Convert DateTime to Cedar expr that can be used in a Cedar policy. */
+    @Override
+    public String toCedarExpr() {
+        return "datetime(\"" + dateTime + "\")";
```
Comment (mark-creamer-amazon) [inline CedarJava/src/main/java/com/cedarpolicy/value/DateTime.java:98]: Worried about this being valid, but should be fine since dateTime is validated on construction, cannot be mutated externally since it's private and it's set once.

Comment (muditchaudhary) [reply CedarJava/src/main/java/com/cedarpolicy/value/DateTime.java:126]: Yeah. It's `final` and only assigned after validation so should be ok. This is what we do for `IpAddress` as well

File: CedarJava/src/main/java/com/cedarpolicy/value/DateTime.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (muditchaudhary) [reply CedarJava/src/main/java/com/cedarpolicy/value/DateTime.java]: Changed to use semantic equality in the revision

**Review by mark-creamer-amazon — APPROVED** [review]
Lgtm
